### PR TITLE
`lazyDefine` components

### PR DESCRIPTION
**Pull request by koddsson**
Upstream a `lazyDefine` function from `github/github` and add some tests.

### CODE REVIEW

File: src/lazy-define.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    for (const el of document.querySelectorAll(tagName)) {
+      observer.observe(el)
+    }
+  })
+
+const strategies: Record<string, Strategy> = {
+  ready: () => ready,
+  firstInteraction: () => firstInteraction,
+  visible
+}
```
Comment (koddsson) [inline src/lazy-define.ts:56]: Move strategies into the `lazyDefine` callbacks in the application code?

File: src/lazy-define.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+export function addStrategy(name: string, strategy: Strategy) {
+  if (name in strategies) {
+    throw new Error(`Strategy ${name} already exists!`)
+  }
+  strategies[name] = strategy
+}
+
```
Comment (koddsson) [inline src/lazy-define.ts:64]:
```suggestion
```

File: test/lazy-define.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    expect(onDefine).to.have.callCount(0)
+    resolve()
+    await promise
+    expect(onDefine).to.be.callCount(1)
+  })
+
+  it("doesn't overwrite a existing strategy", () => {
+    expect(() => addStrategy('ready', spy())).to.throw(/already exists/)
+  })
+})
```
Comment (koddsson) [inline test/lazy-define.ts:30]:
```suggestion
```

File: test/lazy-define.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import {expect, fixture, html} from '@open-wc/testing'
+import {spy} from 'sinon'
+import {addStrategy, lazyDefine} from '../src/lazy-define.js'
```
Comment (koddsson) [inline test/lazy-define.ts:3]:
```suggestion
import {lazyDefine} from '../src/lazy-define.js'
```

**Review by keithamus — APPROVED** [review]

**Comment by keithamus** [comment]
This looks good, we should add some documentation and export it from index :laughing:

**Review by keithamus — APPROVED** [review]

**Pull request merged**